Snapshot publishing is failing

<!-- The bug you're experiencing might have already be reported! -->
<!-- Please search in the [issues](https://github.com/detekt/detekt/issues) before creating one. -->

## Expected Behavior
Snapshots are published from main branch.

## Observed Behavior
https://github.com/detekt/detekt/actions/runs/16289103327/job/45994623450#step:7:597

```
Caused by: org.gradle.internal.resource.transport.http.HttpErrorStatusCodeException: Could not PUT 'https://oss.sonatype.org/content/repositories/snapshots/io/gitlab/arturbosch/detekt/detekt-api/main-SNAPSHOT/detekt-api-main-20250715.091623-2710.jar'. Received status code 403 from server: Forbidden
```

## Steps to Reproduce
Push a commit to main branch.

## Context
Users sometimes rely on snapshots to test before new builds are published.

@cortinico 

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version of detekt used:
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

--- a/.github/workflows/deploy-snapshot.yaml
+++ b/.github/workflows/deploy-snapshot.yaml
@@ -43,4 +43,4 @@ jobs:
           ORG_GRADLE_PROJECT_SIGNING_PWD: ${{ secrets.ORG_GRADLE_PROJECT_SIGNING_PWD }}
           ORG_GRADLE_PROJECT_SONATYPE_USERNAME: ${{ secrets.ORG_GRADLE_PROJECT_SONATYPE_USERNAME }}
           ORG_GRADLE_PROJECT_SONATYPE_PASSWORD: ${{ secrets.ORG_GRADLE_PROJECT_SONATYPE_PASSWORD }}
-        run: ./gradlew publishAllToSonatypeSnapshot -Dsnapshot=true --stacktrace
+        run: ./gradlew publishToSonatype -Dsnapshot=true --stacktrace

--- a/build-logic/src/main/kotlin/packaging.gradle.kts
+++ b/build-logic/src/main/kotlin/packaging.gradle.kts
@@ -5,24 +5,6 @@ plugins {
 }
 
 publishing {
-    repositories {
-        maven {
-            name = "mavenCentral"
-            url = uri("https://oss.sonatype.org/service/local/staging/deploy/maven2")
-            credentials {
-                username = "SONATYPE_USERNAME".byProperty
-                password = "SONATYPE_PASSWORD".byProperty
-            }
-        }
-        maven {
-            name = "sonatypeSnapshot"
-            url = uri("https://oss.sonatype.org/content/repositories/snapshots")
-            credentials {
-                username = "SONATYPE_USERNAME".byProperty
-                password = "SONATYPE_PASSWORD".byProperty
-            }
-        }
-    }
     // We don't need to configure publishing for the Gradle plugin.
     if (project.name != "detekt-gradle-plugin") {
         publications.register<MavenPublication>(DETEKT_PUBLICATION) {

--- a/build-logic/src/main/kotlin/releasing.gradle.kts
+++ b/build-logic/src/main/kotlin/releasing.gradle.kts
@@ -119,23 +119,3 @@ tasks.register("publishToMavenLocal") {
     }
     dependsOn(gradle.includedBuild("detekt-gradle-plugin").task(":publishToMavenLocal"))
 }
-
-tasks.register("publishAllToSonatypeSnapshot") {
-    description = "Publish all the projects to Sonatype Snapshot Repository"
-    subprojects {
-        if (this.plugins.hasPlugin("publishing")) {
-            dependsOn(tasks.named("publishAllPublicationsToSonatypeSnapshotRepository"))
-        }
-    }
-    dependsOn(gradle.includedBuild("detekt-gradle-plugin").task(":publishAllPublicationsToSonatypeSnapshotRepository"))
-}
-
-tasks.register("publishAllToMavenCentral") {
-    description = "Publish all the projects to Sonatype Staging Repository"
-    subprojects {
-        if (this.plugins.hasPlugin("publishing")) {
-            dependsOn(tasks.named("publishAllPublicationsToMavenCentralRepository"))
-        }
-    }
-    dependsOn(gradle.includedBuild("detekt-gradle-plugin").task(":publishAllPublicationsToMavenCentralRepository"))
-}

--- a/scripts/release.sh
+++ b/scripts/release.sh
@@ -2,7 +2,7 @@
 set -e
 gradle publishToMavenLocal
 gradle build
-gradle publishAllToMavenCentral --max-workers 1
+gradle publishToSonatype closeSonatypeStagingRepository --max-workers 1
 gradle :detekt-gradle-plugin:publishPlugins
 gradle githubRelease
 gradle applyDocVersion

